syncer/checkpoint.go

```diff
@@ -439,14 +489,29 @@ func (cp *RemoteCheckPoint) CheckGlobalPoint() bool {
 }
 
 // Rollback implements CheckPoint.Rollback
-func (cp *RemoteCheckPoint) Rollback() {
+func (cp *RemoteCheckPoint) Rollback(schemaTracker *schema.Tracker) {
 	cp.RLock()
 	defer cp.RUnlock()
 	cp.globalPoint.rollback()
 	for schema, mSchema := range cp.points {
 		for table, point := range mSchema {
-			cp.logCtx.L().Info("rollback checkpoint", log.WrapStringerField("checkpoint", point), zap.String("schema", schema), zap.String("table", table))
-			point.rollback()
+			logger := cp.logCtx.L().WithFields(zap.String("schema", schema), zap.String("table", table))
+			logger.Info("rollback checkpoint", log.WrapStringerField("checkpoint", point))
+			if point.rollback() {
+				// schema changed
+				if err := schemaTracker.DropTable(schema, table); err != nil {
+					logger.Debug("failed to drop table from schema tracker", log.ShortError(err))
+				}
+				if point.ti != nil {
+					// TODO: Figure out how to recover from errors.
+					if err := schemaTracker.CreateSchemaIfNotExists(schema); err != nil {
+						logger.Error("failed to rollback schema on schema tracker: cannot create schema", log.ShortError(err))
```

Comment:
how about add the schema name in the error message?

Reply:
The schema name is already included in line 498.